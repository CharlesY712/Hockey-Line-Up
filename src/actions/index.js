export const selectSeason = (season) => ({
  type: 'SELECT_SEASON',
  season
});