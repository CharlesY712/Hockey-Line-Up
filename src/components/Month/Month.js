import React from 'react';

const Month = () => {
  return (
    <div>MONTH</div>
  );
};

export default Month;