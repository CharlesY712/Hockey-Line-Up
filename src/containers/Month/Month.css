.date,
.directions {
  text-align: center;
  color: #677584;
}

.loading-gif {
  width: 100vw;
  height: 80vh;
}