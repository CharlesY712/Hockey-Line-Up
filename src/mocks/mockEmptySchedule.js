export const emptySchedule = {
  "dailygameschedule": {
    "lastUpdatedOn": "2018-04-15 11:54:02 PM"
  }
};