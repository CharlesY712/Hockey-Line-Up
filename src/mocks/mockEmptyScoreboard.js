export const emptyScoreboard = {
  "scoreboard": {
    "lastUpdatedOn": "2018-04-12 1:56:15 PM"
  }
};