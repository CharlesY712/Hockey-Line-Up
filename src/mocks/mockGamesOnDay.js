export const GamesOnDay = [
  {
    "id": "32629",
    "scheduleStatus": "Normal",
    "originalDate": null,
    "originalTime": null,
    "delayedOrPostponedReason": null,
    "date": "2016-10-12",
    "time": "7:00PM",
    "awayTeam": {
      "ID": "12",
      "City": "Toronto",
      "Name": "Maple Leafs",
      "Abbreviation": "TOR"
    },
    "homeTeam": {
      "ID": "13",
      "City": "Ottawa",
      "Name": "Senators",
      "Abbreviation": "OTT"
    },
    "location": "Canadian Tire Centre"
  },
  {
    "id": "32630",
    "scheduleStatus": "Normal",
    "originalDate": null,
    "originalTime": null,
    "delayedOrPostponedReason": null,
    "date": "2016-10-12",
    "time": "8:00PM",
    "awayTeam": {
      "ID": "17",
      "City": "St. Louis",
      "Name": "Blues",
      "Abbreviation": "STL"
    },
    "homeTeam": {
      "ID": "20",
      "City": "Chicago",
      "Name": "Blackhawks",
      "Abbreviation": "CHI"
    },
    "location": "United Center"
  },
  {
    "id": "32631",
    "scheduleStatus": "Normal",
    "originalDate": null,
    "originalTime": null,
    "delayedOrPostponedReason": null,
    "date": "2016-10-12",
    "time": "10:00PM",
    "awayTeam": {
      "ID": "23",
      "City": "Calgary",
      "Name": "Flames",
      "Abbreviation": "CGY"
    },
    "homeTeam": {
      "ID": "24",
      "City": "Edmonton",
      "Name": "Oilers",
      "Abbreviation": "EDM"
    },
    "location": "Rogers Place"
  },
  {
    "id": "32632",
    "scheduleStatus": "Normal",
    "originalDate": null,
    "originalTime": null,
    "delayedOrPostponedReason": null,
    "date": "2016-10-12",
    "time": "10:30PM",
    "awayTeam": {
      "ID": "28",
      "City": "Los Angeles",
      "Name": "Kings",
      "Abbreviation": "LAK"
    },
    "homeTeam": {
      "ID": "26",
      "City": "San Jose",
      "Name": "Sharks",
      "Abbreviation": "SJS"
    },
    "location": "SAP Center at San Jose"
  }
];
