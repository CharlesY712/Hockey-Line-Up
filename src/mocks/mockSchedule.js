export const schedule = {
  "dailygameschedule": {
    "lastUpdatedOn": "2018-04-12 4:24:49 PM",
    "gameentry": [
      {
        "id": "45866",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "3:00PM",
        "awayTeam": {
          "ID": "22",
          "City": "Colorado",
          "Name": "Avalanche",
          "Abbreviation": "COL"
        },
        "homeTeam": {
          "ID": "18",
          "City": "Nashville",
          "Name": "Predators",
          "Abbreviation": "NSH"
        },
        "location": "Bridgestone Arena"
      },
      {
        "id": "45875",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "3:00PM",
        "awayTeam": {
          "ID": "7",
          "City": "New Jersey",
          "Name": "Devils",
          "Abbreviation": "NJD"
        },
        "homeTeam": {
          "ID": "1",
          "City": "Tampa Bay",
          "Name": "Lightning",
          "Abbreviation": "TBL"
        },
        "location": "Amalie Arena"
      },
      {
        "id": "45874",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "8:00PM",
        "awayTeam": {
          "ID": "12",
          "City": "Toronto",
          "Name": "Maple Leafs",
          "Abbreviation": "TOR"
        },
        "homeTeam": {
          "ID": "11",
          "City": "Boston",
          "Name": "Bruins",
          "Abbreviation": "BOS"
        },
        "location": "TD Garden"
      },
      {
        "id": "45873",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "10:30PM",
        "awayTeam": {
          "ID": "26",
          "City": "San Jose",
          "Name": "Sharks",
          "Abbreviation": "SJS"
        },
        "homeTeam": {
          "ID": "29",
          "City": "Anaheim",
          "Name": "Ducks",
          "Abbreviation": "ANA"
        },
        "location": "Honda Center"
      }
    ]
  }
};