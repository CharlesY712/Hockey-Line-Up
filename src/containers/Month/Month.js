import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Game from '../../components/Game/Game';
import * as actions from '../../actions';
import { fetchSeason } from '../../helpers/fetchSeason';
import loadingGif from '../../images/icons/blue_loading.gif';
import './Month.css';

class Month extends Component {
  constructor() {
    super();
    this.state = {
      isLoading: false,
      games: []
    };
  }

  componentDidMount() {
    if (this.props.date.length === 7) {
      this.getMonthDays();
      this.setState({isLoading: true});
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.date !== this.props.date) {
      if (this.props.date.length === 7) {
        this.setState({isLoading: true});
        this.getMonthDays();
      }
    }
  }

  async getMonthDays() {
    const date = this.props.date + '-01';
    let monthOfGames = [];
    for (let index = 0; index < 28; index++) {
      let myDate = new Date(date);
      myDate.setDate(myDate.getDate() + index);
      let nextDay = myDate.toISOString().slice(0, 10);
      const games = await fetchSeason(nextDay);
      const gameSchedule = games.dailygameschedule.gameentry;
      if (gameSchedule !== undefined) {
        monthOfGames = [...monthOfGames, ...gameSchedule];
      }
    }
    this.props.addSchedule(monthOfGames);
    const gameComponents = this.displayGames(monthOfGames);
    this.setState({games: gameComponents, isLoading: false});
  }

  displayGames() {
    const mappedGames = this.props.schedule.map(schedule => {
      return <Game
        key={schedule.id}
        homeTeamCity={schedule.homeTeam.City}
        homeTeamName={schedule.homeTeam.Name}
        awayTeamCity={schedule.awayTeam.City}
        awayTeamName={schedule.awayTeam.Name}
        time={schedule.time}
        date={schedule.date.slice(5)}
      />;
    });
    return mappedGames;
  }
  
  render() {
    if (this.props.date.length === 7) {
      return (
        <div>
          {
            this.state.isLoading &&   
        <section>
          <img className="loading-gif" src={loadingGif} alt="loading"/>
        </section>
          }
          {
            !this.state.isLoading &&   
          <section>
            <div className="directions">Please select a Month in the box above.</div>
            <h1 className="date">{this.props.date}</h1>
            {this.state.games}
          </section>
          }
        </div>
      );
    } else {
      return (
        <section>
          <div className="directions">Please select a Month in the box above.</div>
        </section>
      );
    }
  }
}

Month.propTypes = {
  addSchedule: PropTypes.func,
  addScoreboard: PropTypes.func,
  date: PropTypes.string,
  setDate: PropTypes.func,
  schedule: PropTypes.array
};

export const mapStateToProps = state => ({
  date: state.setDate,
  schedule: state.schedule
});

export const mapDispatchToProps = dispatch => ({
  addSchedule: (season) => dispatch(actions.addSchedule(season)),
  setDate: (date) => dispatch(actions.setDate(date))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Month));
