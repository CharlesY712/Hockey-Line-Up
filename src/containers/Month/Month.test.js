import { mapStateToProps, mapDispatchToProps } from './Month';
import { schedule }  from '../../mocks/mockSchedule';
import * as actions from '../../actions';

describe('Month', () => {

  describe('Month container', () => {

  });

  describe('mapStateToProps', () => {
    
    it('correctly maps the schedule to props', () => {
      const mockState = { schedule: [schedule] };
      const mapped = mapStateToProps(mockState);
      const expected = [schedule];

      expect(mapped.schedule).toEqual(expected);
    });

    it('correctly maps the date to props', () => {
      const date = '2018-04-12';
      const mockState = { setDate: date };
      const mapped = mapStateToProps(mockState);
      const expected = date;

      expect(mapped.date).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct params on addSchedule', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      const expected = actions.addSchedule(schedule);

      mapped.addSchedule(schedule);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with the correct params on setDate', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      const expected = actions.setDate(schedule);

      mapped.setDate(schedule);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });
  });
});