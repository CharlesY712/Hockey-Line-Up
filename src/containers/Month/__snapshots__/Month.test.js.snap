// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Month container should display a loading gif if state.isLoading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Month
    addSchedule={[Function]}
    date="2018-04"
    schedule={
        Array [
            Object {
              "awayTeam": Object {
                "Abbreviation": "COL",
                "City": "Colorado",
                "ID": "22",
                "Name": "Avalanche",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "NSH",
                "City": "Nashville",
                "ID": "18",
                "Name": "Predators",
              },
              "id": "45866",
              "location": "Bridgestone Arena",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "3:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "NJD",
                "City": "New Jersey",
                "ID": "7",
                "Name": "Devils",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "TBL",
                "City": "Tampa Bay",
                "ID": "1",
                "Name": "Lightning",
              },
              "id": "45875",
              "location": "Amalie Arena",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "3:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "TOR",
                "City": "Toronto",
                "ID": "12",
                "Name": "Maple Leafs",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "BOS",
                "City": "Boston",
                "ID": "11",
                "Name": "Bruins",
              },
              "id": "45874",
              "location": "TD Garden",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "8:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "SJS",
                "City": "San Jose",
                "ID": "26",
                "Name": "Sharks",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "ANA",
                "City": "Anaheim",
                "ID": "29",
                "Name": "Ducks",
              },
              "id": "45873",
              "location": "Honda Center",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "10:30PM",
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section>
          <img
                    alt="loading"
                    className="loading-gif"
                    src="blue_loading.gif"
          />
</section>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="loading"
            className="loading-gif"
            src="blue_loading.gif"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading",
            "className": "loading-gif",
            "src": "blue_loading.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "section",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section>
            <img
                        alt="loading"
                        className="loading-gif"
                        src="blue_loading.gif"
            />
</section>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="loading"
              className="loading-gif"
              src="blue_loading.gif"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "loading",
              "className": "loading-gif",
              "src": "blue_loading.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "section",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Month Month container should not display games on mount if date length is not 7 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Month
    addSchedule={[Function]}
    date="2018-04-12"
    schedule={
        Array [
            Object {
              "dailygameschedule": Object {
                "gameentry": Array [
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "COL",
                      "City": "Colorado",
                      "ID": "22",
                      "Name": "Avalanche",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "NSH",
                      "City": "Nashville",
                      "ID": "18",
                      "Name": "Predators",
                    },
                    "id": "45866",
                    "location": "Bridgestone Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "NJD",
                      "City": "New Jersey",
                      "ID": "7",
                      "Name": "Devils",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "TBL",
                      "City": "Tampa Bay",
                      "ID": "1",
                      "Name": "Lightning",
                    },
                    "id": "45875",
                    "location": "Amalie Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "TOR",
                      "City": "Toronto",
                      "ID": "12",
                      "Name": "Maple Leafs",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "BOS",
                      "City": "Boston",
                      "ID": "11",
                      "Name": "Bruins",
                    },
                    "id": "45874",
                    "location": "TD Garden",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "8:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "SJS",
                      "City": "San Jose",
                      "ID": "26",
                      "Name": "Sharks",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "ANA",
                      "City": "Anaheim",
                      "ID": "29",
                      "Name": "Ducks",
                    },
                    "id": "45873",
                    "location": "Honda Center",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "10:30PM",
                  },
                ],
                "lastUpdatedOn": "2018-04-12 4:24:49 PM",
              },
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="directions"
>
        Please select a Month in the box above.
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please select a Month in the box above.",
        "className": "directions",
      },
      "ref": null,
      "rendered": "Please select a Month in the box above.",
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="directions"
>
          Please select a Month in the box above.
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select a Month in the box above.",
          "className": "directions",
        },
        "ref": null,
        "rendered": "Please select a Month in the box above.",
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Month Month container should not display games on update if date length is not 7 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Month
    addSchedule={[Function]}
    date="2018-04-12"
    schedule={
        Array [
            Object {
              "dailygameschedule": Object {
                "gameentry": Array [
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "COL",
                      "City": "Colorado",
                      "ID": "22",
                      "Name": "Avalanche",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "NSH",
                      "City": "Nashville",
                      "ID": "18",
                      "Name": "Predators",
                    },
                    "id": "45866",
                    "location": "Bridgestone Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "NJD",
                      "City": "New Jersey",
                      "ID": "7",
                      "Name": "Devils",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "TBL",
                      "City": "Tampa Bay",
                      "ID": "1",
                      "Name": "Lightning",
                    },
                    "id": "45875",
                    "location": "Amalie Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "TOR",
                      "City": "Toronto",
                      "ID": "12",
                      "Name": "Maple Leafs",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "BOS",
                      "City": "Boston",
                      "ID": "11",
                      "Name": "Bruins",
                    },
                    "id": "45874",
                    "location": "TD Garden",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "8:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "SJS",
                      "City": "San Jose",
                      "ID": "26",
                      "Name": "Sharks",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "ANA",
                      "City": "Anaheim",
                      "ID": "29",
                      "Name": "Ducks",
                    },
                    "id": "45873",
                    "location": "Honda Center",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "10:30PM",
                  },
                ],
                "lastUpdatedOn": "2018-04-12 4:24:49 PM",
              },
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="directions"
>
        Please select a Month in the box above.
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please select a Month in the box above.",
        "className": "directions",
      },
      "ref": null,
      "rendered": "Please select a Month in the box above.",
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="directions"
>
          Please select a Month in the box above.
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select a Month in the box above.",
          "className": "directions",
        },
        "ref": null,
        "rendered": "Please select a Month in the box above.",
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Month Month container should not display schedule if there are no games in Month 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Month
    addSchedule={[Function]}
    date="2018-06"
    schedule={
        Array [
            Object {
              "awayTeam": Object {
                "Abbreviation": "COL",
                "City": "Colorado",
                "ID": "22",
                "Name": "Avalanche",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "NSH",
                "City": "Nashville",
                "ID": "18",
                "Name": "Predators",
              },
              "id": "45866",
              "location": "Bridgestone Arena",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "3:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "NJD",
                "City": "New Jersey",
                "ID": "7",
                "Name": "Devils",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "TBL",
                "City": "Tampa Bay",
                "ID": "1",
                "Name": "Lightning",
              },
              "id": "45875",
              "location": "Amalie Arena",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "3:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "TOR",
                "City": "Toronto",
                "ID": "12",
                "Name": "Maple Leafs",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "BOS",
                "City": "Boston",
                "ID": "11",
                "Name": "Bruins",
              },
              "id": "45874",
              "location": "TD Garden",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "8:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "SJS",
                "City": "San Jose",
                "ID": "26",
                "Name": "Sharks",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "ANA",
                "City": "Anaheim",
                "ID": "29",
                "Name": "Ducks",
              },
              "id": "45873",
              "location": "Honda Center",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "10:30PM",
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section>
          <img
                    alt="loading"
                    className="loading-gif"
                    src="blue_loading.gif"
          />
</section>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="loading"
            className="loading-gif"
            src="blue_loading.gif"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading",
            "className": "loading-gif",
            "src": "blue_loading.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "section",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section>
            <img
                        alt="loading"
                        className="loading-gif"
                        src="blue_loading.gif"
            />
</section>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="loading"
              className="loading-gif"
              src="blue_loading.gif"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "loading",
              "className": "loading-gif",
              "src": "blue_loading.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "section",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
